src/_pytest/config/__init__.py

```diff
@@ -899,6 +900,20 @@ class InvocationParams:
         dir: Path
         """The directory from which :func:`pytest.main` was invoked."""
 
+    @final
+    class ArgsSource(enum.IntEnum):
```

Comment:
```suggestion
    class ArgsSource(enum.Enum):
```

since, contrary to `ExitStatus`, I see no reason why `ArgsSource` members would be used as integers (see [Python docs](https://docs.python.org/3/library/enum.html#intenum))